lib/auth_processes/mongodb.js

```diff
@@ -1,383 +1,142 @@
 'use strict';
```

Comment:
Lets make a ticket to move this into `mongodb-core`

Reply:
[NODE-1601](https://jira.mongodb.org/browse/NODE-1601)